library/alloc/src/vec/mod.rs

```diff
@@ -1912,6 +1918,14 @@ impl<T, A: Allocator> Vec<T, A> {
     /// vec.push(3);
     /// assert_eq!(vec, [1, 2, 3]);
     /// ```
+    ///
+    /// # Time complexity
+    ///
+    /// Takes amortized *O*(1) time. If the vector's length would exceed its
+    /// capacity after the push, the capacity is doubled by allocating
```

Comment:
While true *for now*, I don't think doubling is something we want to promise in documentation, just amortized growth. For example, some languages only double up to a certain size and then use a smaller growth factor.

Reply:
Good point! I'll make it more vague

Reply:
What do you think of this?
```rs
    /// Takes amortized *O*(1) time. If the vector's length would exceed its
    /// capacity after the push, the capacity is increased and *O*(*capacity*)
    /// time is taken to copy the vector's elements. This expensive operation is
    /// offset by the *capacity* *O*(1) insertions it allows.
```